### ISSUE TITLE
Caching field normalize and bigfield self_reduce

### ISSUE THREAD

**Issue by Rumata888** [AztecProtocol/barretenberg#1052]
Currently we just add constraints and return new elements when we normalize field elements and reduce bigfield elements. This means that a call to the same field element twice to normalize it can produce 2 sets of identical constraints. Ideally, we would cache the result and just return it

**Comment by elijahhampton** [comment]
Is this issue open to work? @Rumata888 

I hesitated due to it being opened in the mirrored repository vs aztec-packages

**Comment by ledwards2225** [comment]
@suyash67 Did you guys discuss this issue and decide against?

**Comment by suyash67** [comment]
Yes. Although this could be very useful, but adding such a flag can be very error-prone, especially at this stage. Hence, closing.
